clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/validators/servergroup/CloneOpenstackAtomicOperationValidator.groovy

```diff
@@ -35,39 +35,45 @@ class CloneOpenstackAtomicOperationValidator extends DescriptionValidator<CloneO
 
   @Override
   void validate(List priorDescriptions, CloneOpenstackAtomicOperationDescription description, Errors errors) {
-    def helper = new OpenstackAttributeValidator("cloneOpenstackAtomicOperationDescription", errors)
+    def validator = new OpenstackAttributeValidator("cloneOpenstackAtomicOperationDescription", errors)
 
-    if (!helper.validateCredentials(description.account, accountCredentialsProvider)) {
+    if (!validator.validateCredentials(description.account, accountCredentialsProvider)) {
       return
     }
-
-    OpenstackCredentials credentials = (OpenstackCredentials) accountCredentialsProvider.getCredentials(description.account).credentials
-
-    helper.validateServerGroupCloneSource(description.source, "source")
-
-    if (description.application) {
-      helper.validateApplication(description.application, "application")
+    if (!validator.validateNotNull(description.source, "source")) {
+      return
     }
-
-    if (description.stack) {
-      helper.validateStack(description.stack, "stack")
+    validator.validateNotEmpty(description.source.region, "region")
+    validator.validateNotEmpty(description.source.serverGroup, "serverGroup")
+    if (description.application) {
+      validator.validateApplication(description.application, "application")
     }
-
-    if (description.freeFormDetails) {
-      helper.validateDetails(description.freeFormDetails, "details")
+    validator.validateStack(description.stack, "stack")
+    validator.validateDetails(description.freeFormDetails, "details")
+    if (description.serverGroupParameters) {
+      validateServerGroup(validator, description.serverGroupParameters)
     }
-
-    if (description.region) {
-      helper.validateNotEmpty(description.region, "region")
+    if (description.timeoutMins) {
+      validator.validateNonNegative(description.timeoutMins, "timeoutMins")
     }
+  }
 
-    if (description.heatTemplate) {
-      helper.validateHeatTemplate(description.heatTemplate, "heatTemplate", accountCredentialsProvider, description.account)
-    }
 
-    if (description.timeoutMins) {
-      helper.validateNonNegative(description.timeoutMins, "timeoutMins")
+  //TODO this is copy-paste from DeployOpenstackAtomicOperationValidator
+  def validateServerGroup(OpenstackAttributeValidator validator, ServerGroupParameters parameters) {
+    String prefix = "serverGroupParameters"
+    parameters.with {
+      validator.validateNotEmpty(instanceType, "${prefix}.instanceType")
+      validator.validateNotEmpty(image, "${prefix}.image")
+      validator.validateNotNull(maxSize, "${prefix}.maxSize")
+      validator.validatePositive(maxSize, "${prefix}.maxSize")
```

Comment:
Seems like your `validatePositive` now also checks that the value isn't `null`, so those `NotNull` checks seem superfluous